client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import {
+	MEDIA_FILE_UPLOAD_FAILURE,
+	MEDIA_FILE_UPLOAD_SUCCESS,
+	MEDIA_FILE_UPLOADS_ENQUEUE
+} from 'state/action-types';
+import { uploadFile } from 'state/media/actions';
+import { isMediaUploadInProgress, getNextQueuedUpload } from 'state/media/selectors';
+
+export function uploadNext( store ) {
+	const nextQueued = getNextQueuedUpload( store.getState() );
+	if ( nextQueued ) {
+		const { siteId, file } = nextQueued;
+		store.dispatch( uploadFile( siteId, file ) );
+	}
+}
+
+export function maybeUploadFirst( store, action ) {
+	if ( ! isMediaUploadInProgress( store.getState() ) ) {
```

Comment:
>Not sure if this is being used elsewhere, but if the value of whether or not we are currently uploading is really contained to this file, does it need to be stored in Redux?

Looking at the current UI requirements, we'll need some upload indicators managed in state, but specifically for the individual items themselves. At the moment, we're not using the queue in any form. To answer your question, no, I don't expect any persistence between sessions, and this will only ever occur in the browser now. That said, I'm wary of module-level variables existing in state, given the SSR use-case. Seems unlikely that we'll need this for SSR anytime in the near future, but I don't want to set a potentially problematic precedent for others to follow.